state/unit.go

```diff
@@ -2804,10 +2804,25 @@ type ActionSpecsByName map[string]charm.ActionSpec
 // AddAction adds a new Action of type name and using arguments payload to
 // this Unit, and returns its ID.  Note that the use of spec.InsertDefaults
 // mutates payload.
-func (u *Unit) AddAction(operationID, name string, payload map[string]interface{}) (Action, error) {
+func (u *Unit) AddAction(operationID, name string, payload map[string]interface{}) (_ Action, err error) {
 	if len(name) == 0 {
```

Comment:
It might be worth to wrap this into a func then it can be called from here and machine.AddAction.